backend/cms/views/pages/page_form.py

```diff
@@ -19,109 +39,155 @@ class PageForm(forms.ModelForm):
-            )
-            page.move_to(self.cleaned_data['parent'], self.cleaned_data['position'])
-
-        if page_translation:
-            # save page translation
-            page_translation.slug = slug
-            page_translation.title = self.cleaned_data['title']
-            page_translation.text = self.cleaned_data['text']
-            page_translation.status = self.cleaned_data['status']
-            page_translation.save()
-        else:
-            # create page translation
-            page_translation = PageTranslation.objects.create(
-                slug=slug,
-                title=self.cleaned_data['title'],
-                text=self.cleaned_data['text'],
-                status=self.cleaned_data['status'],
-                language=Language.objects.get(code=language_code),
-                public=self.cleaned_data['public'],
-                page=page,
-                creator=self.user
-            )
+        logger.info('New PageTranslationForm with args {} and kwargs {}'.format(args, kwargs))
 
-        return page
+        # to set the public value through the submit button, we have to overwrite the field value for public.
+        # we could also do this in the save() function, but this would mean that it is not recognized in changed_data.
+        # check if POST data was submitted and the publish button was pressed
+        if len(args) == 1 and 'submit_publish' in args[0]:
+            # copy QueryDict because it is immutable
+            post = args[0].copy()
+            # remove the old public value (might be False and update() does only append, not overwrite)
+            post.pop('public')
+            # update the POST field with True (has to be a string to make sure the field is recognized as changed)
+            post.update({'public': 'True'})
+            # set the args to POST again
+            args = (post,)
+            logger.info('changed POST arg public manually to True')
+
+        super(PageTranslationForm, self).__init__(*args, **kwargs)
+
+        self.fields['public'].widget = forms.Select(choices=self.PUBLIC_CHOICES)
+
+    def save(self, *args, **kwargs):
+
+        logger.info('PageTranslationForm saved with args {} and kwargs {}'.format(args, kwargs))
+
+        # pop kwarg to make sure the super class does not get this param
+        page = kwargs.pop('page', None)
+        language = kwargs.pop('language', None)
+        user = kwargs.pop('user', None)
+
+        if not self.instance.id:
+            # don't commit saving of ModelForm, because required fields are still missing
+            kwargs['commit'] = False
+
+        page_translation = super(PageTranslationForm, self).save(*args, **kwargs)
+
+        if not self.instance.id:
+            # only update these values when page translation is created
+            page_translation.page = page
+            page_translation.creator = user
+            page_translation.language = language
+
+        if not self.cleaned_data['slug']:
```

Comment:
One thing I noticed here:
If the slug is present and changed, we have to check if it is unique per region and language.

Reply:
resolved in the new version of the commit.